packages/clerk-js/src/ui/components/UserProfile/PhoneSection.tsx

```diff
@@ -35,9 +35,18 @@ const PhoneScreen = (props: PhoneScreenProps) => {
   );
 };
 
-export const PhoneSection = () => {
+export const PhoneSection = ({ shouldAllowCreation = true }: { shouldAllowCreation?: boolean }) => {
   const { user } = useUser();
 
+  /**
+   * Does not render if:
+   * - User is not present
+   * - Creation isn't allowed AND there aren't external accounts
+   */
+  if (!user || (!shouldAllowCreation && user.phoneNumbers.length === 0)) {
+    return null;
+  }
```

Comment:
```suggestion
  /**
   * Does not render if creation isn't allowed AND there aren't external accounts
   */
  if (!shouldAllowCreation && user.phoneNumbers.length === 0) {
    return null;
  }
```

Reply:
Do you think the comment still valid as the condition is more simple now? Perhaps I can extract to a variable to make it easier to follow

Reply:
Let's keep `shouldAllowCreation` since it's semantic enought, and extract `user.phoneNumbers.length > 0` to `hasExternalAccounts`:

```ts
if (!shouldAllowCreation && !hasExternalAccounts) { 
```

Reply:
sound good!

Reply:
Resolved by putting the condition inside a variable